core/src/main/java/org/apache/calcite/sql/SqlKind.java

```diff
@@ -1382,6 +1382,17 @@ public SqlKind negateNullSafe() {
     }
   }
 
+  public SqlKind negateNullSafe2() {
```

Comment:
What does `2` mean here? The existing `negate` and `negateNullSafe` are already confusing, and `negateNullSafe2` makes it even more confusing.

Why this method considers only `IS_NOT_NULL` and `IS_NULL`? Why does it take `IS_TRUE` from `negateNullSafe` rather than from `negate`?